File: backend/src/003-archivos/read_write/read_write.js
"use strict"

var fs = require("fs");

// Abrir un archivo para escritura
fs.open("ejemplo.txt", "w", (error, archivo) => {
    if (error) {
        if (error.code === "EEXIST") {
            console.error("El archivo ya existe");
            return;
        } else {
            throw error;
        }
    }

    // Escribir en archivo
    fs.write(archivo, "Contenido del archivo\n");
    fs.close(archivo);
})


// Leer el contenido de un archivo
fs.readFile("ejemplo.txt", "utf8", (error, contenido) => {
    if (error) {
        throw error;
    } else {
        console.log(contenido);
    }
});

File: backend/src/003-archivos/read_write/read_write.c
#include <stdio.h>

int main() {
    FILE *manejador;
    FILE *manejador2;

    // Abrir un archivo para escritura
    manejador = fopen("ejemplo.txt", "w");
    if (manejador != 0) {
        printf("Archivo abierto para escritura.\n");
        // Escribir en un archivo
        fprintf(manejador, "%s\n", "Contenido del archivo.");
        fclose(manejador);
    } else {
        printf("Ocurrió un error al intentar escribir el archivo.\n");
        if (manejador != NULL) {
            fclose(manejador);
        }
        return 1;
    }

    // Abrir un archivo para lectura
    char buffer[1024];
    manejador2 = fopen("ejemplo.txt", "r");
    if (manejador2 != 0) {
        printf("Archivo abierto para lectura.\n");
        // Leer la primera línea del archivo
        fscanf(manejador2, "%[^\n]", buffer);
        printf("%s\n", buffer);
    } else {
        printf("Ocurrió un error al intentar leer el archivo.\n");
        if (manejador2 != NULL) {
            fclose(manejador2);
        }
        return 1;
    }

    return 0;
}

/* Fuentes:
https://code-reference.com/c/stdio.h/fopen
https://stackoverflow.com/questions/11573974/write-to-txt-file
https://www.programiz.com/c-programming/examples/read-file
*/